tests/utils/logging_tester.py

```diff
@@ -0,0 +1,79 @@
+from functools import wraps
+import os
+import json
+
+
+def _check_logs(reference_logfile, generated_logfile):
+    reference_logs = open(reference_logfile, "r").readlines()
+    generated_logs = open(generated_logfile, "r").readlines()
+
+    if len(reference_logs) != len(generated_logs):
+        msg_assert = ""
+        if len(reference_logs) > len(generated_logs):
+            diff = len(reference_logs) - len(generated_logs)
+            msg_assert += f"Logs generated are SHORTER than the expected logs. There are {diff} extra logs.\n"
+            msg_assert += "Last log of the generated log is : \n"
+            msg_assert += generated_logs[-1]
+        else:
+            diff = len(generated_logs) - len(reference_logs)
+            msg_assert += f"Logs generated are LONGER than the expected logs.\n There are {diff} extra logs :\n"
+            for log in generated_logs[len(reference_logs) :]:
+                msg_assert += log
```

Comment:
Comment. I'm not loving the asymmetry here. Why output only the last line in one case but all lines in the other?

Reply:
Also, why not use a regular diff? I understand they can be long and hard to read.... are there any other reasons?

Reply:
Yes, the main reason is that they _are_ often long and hard to read. Furthermore, the whole point of having json formatted logs is be able to do pretty differences display 👍🏻 

Concerning the asymetry, I've been thunking of this and I think it's better. Like, when generated logs are shorter than what expected, it's better to see what is the last log generated than showing all the logs that should have been there, as these can be accessed in the expected logs.

Reply:
I'm not convinced that one is necessarily better than the other in one case or another. But you have more experience in debugging test logs so you should have final sya.